programs/survival/waystones.sc

```diff
@@ -0,0 +1,310 @@
+// This app makes lodestones into waystones players can teleport to.
+// The goal is to add waystones to the game in a balanced and natural feeling way.
+
+// When a player clicks a lodestone it displays a list of
+// waystones the player has perviously visited.
+// the icon for each location is the block the lodestone was placed on.
+// Waystone can be named by naming the lodestone in an anvil or with a name tag
+// after the fact.
+
+// Configurable Features:
+// Teleporting to a waystone costs an ender_eye
+// A players waypoints get reset on death.
+// Activating a waystone requires 6 copper blocks in a 5x5 area around the lodestone
+// by default waystones only work in the overworld dimension.
+// by default waystones can't cross dimensions.
+
+// Missing features,
+// - pagination, currently the app will display 54 waypoints per player.
+
+
+global_settings = {
+    'dimensional_crossing'->false,
+    'enabled_dimensions'->'overworld',
+    'offering'->'ender_eye',
+    'clear_waypoints_on_death'->true,
+    'structure_material'->'copper',
+    'structure_size'->6
+};
+
+__config()->{
+    'command_permission'->'ops',
+    'scope'->'global',
+    'commands'->{
+        ''->'open_waystones_screen',
+        'setting offering <item>'->['setting_command','offering'],
+        'setting offering'->['setting_read','offering'],
+        'setting dimensionalCrossing <bool>'->['setting_command','dimensional_crossing'],
+        'setting dimensionalCrossing'->['setting_read','dimensional_crossing'],
+        'setting clearWaypointsOnDeath <bool>'->['setting_command','clear_waypoints_on_death'],
+        'setting clearWaypointsOnDeath'->['setting_read','clear_waypoints_on_death'],
+        'setting enabledDimensions <dimensions>'->['setting_command','enabled_dimensions'],
+        'setting enabledDimensions'->['setting_read','enabled_dimensions'],
+        'setting structureMaterial <dimensions>'->['setting_command','structure_material'],
```

Comment:
Missed this

Reply:
Woops

Reply:
i changed it to term instead of block to allow for less strict inputs like `plank`, `copper`, `log`, `birch` instead of strict block names.